repose-aggregator/components/filters/valkyrie-authorization/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala

```diff
@@ -135,6 +137,47 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
       }
     }
 
+    def getInventory(headerResult: ValkyrieResult): ValkyrieResult = {
+      def parseInventory(inputStream: InputStream): Try[UserPermissions] = {
+
+        @tailrec
+        def parseJson(deviceToPermissions: List[DeviceToPermission], values: List[JsValue]): UserPermissions = {
+          if (values.isEmpty) {
+            UserPermissions(List(ACCOUNT_ADMIN).toVector, deviceToPermissions.toVector)
+          } else {
+            val currentItem: JsValue = values.head
+            (currentItem \ "id").as[Int] match {
+              case id if id > 0 =>
+                parseJson(DeviceToPermission(id, ACCOUNT_ADMIN) +: deviceToPermissions, values.tail)
+              case _ => parseJson(deviceToPermissions, values.tail)
+            }
+          }
+        }
+
+        val input: String = Source.fromInputStream(inputStream).getLines() mkString ""
+        try {
+          val json = Json.parse(input)
+          val inventory: List[JsValue] = (json \ "inventory").as[List[JsValue]]
+          Success(parseJson(List.empty[DeviceToPermission], inventory))
+        } catch {
+          case e: Exception =>
+            logger.error(s"Invalid Json response from Valkyrie: $input", e)
+            Failure(new Exception("Invalid Json response from Valkyrie", e))
+        }
+      }
+
+      headerResult match {
+        case UserInfo(tenant, contact) =>
+          //  authorize device            || cull list                  || translate account permissions
+          if (requestedDeviceId.isDefined || matchingResources.nonEmpty || translateAccountPermissions.isDefined) {
+            datastoreValue(tenant, contact, ACCOUNT_ADMIN, configuration.getValkyrieServer, _.asInstanceOf[UserPermissions], parseInventory, tracingHeader)
+          } else {
+            ResponseResult(200)
```

Comment:
I think this line is unreachable.  The if check is already done in the getPermissions method, and if it fails there then the returned ResponseResult(200) would prevent this method from ever being called.